engine/crates/runtime-local/src/operation_cache.rs

```diff
@@ -49,8 +50,10 @@ impl<V> OperationCache<V> for InMemoryOperationCache<V>
 where
     V: Clone + Send + Sync + 'static + serde::Serialize + serde::de::DeserializeOwned,
 {
-    async fn insert(&self, key: String, value: V) {
+    async fn insert(&self, key: String, value: V) -> u64 {
         self.inner.insert(key, value);
+        self.inner.sync();
```

Comment:
I wonder if that's expensive. I guess we'll see.

Reply:
we run this outside from the main thread anyhow. this is the only way of getting the real counter value right after adding a thing to the cache.